src/controllers/company.number.controller.ts

```diff
@@ -14,20 +17,28 @@ import logger from "../utils/logger";
  * @param next
  */
 const route = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
-    logger.debugRequest(req, "Attempt to retrieve and show the company details");
+  logger.debugRequest(req, "Attempt to retrieve and show the company details");
 
-    const companyNumber: string = req.body.companyNumber;
-    const session: Session = req.session as Session;
-    logger.infoRequest(req, `Retrieving company profile for company number ${companyNumber}`);
-    const token: string = getValidAccessToken(session) as string;
+  const companyNumber: string = req.body.companyNumber;
+  const session: Session = req.session as Session;
+  logger.infoRequest(req, `Retrieving company profile for company number ${companyNumber}`);
 
-    const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+  let token: string;
+  try {
+    token = retrieveAccessTokenFromSession(session);
+  } catch (e) {
+    return next(e);
+  }
 
-    if (session) {
-        createObjectionsSession(session);
-        addToObjectionsSession(session, SESSION_COMPANY_PROFILE, company);
-    }
+  const company: ObjectionCompanyProfile = await getCompanyProfile(companyNumber, token);
+
+  const objectionsSessionExtraData: ObjectionSessionExtraData = retrieveObjectionsSessionFromSession(session);
+  if (objectionsSessionExtraData) {
+    objectionsSessionExtraData.objections_company_profile = company;
```

Comment:
could you make objections_company_profile a constant, possibly in constants.ts - i did think to create an enum inside constant.ts for the session value constants but I think it should at least be a constant.

Reply:
It already is, I forgot to use the constant here though, will update

Reply:
thinking some more - would it be better to move this block of code into objections.session.service and have a function like addCompanyProfileToSession

Reply:
> It already is, I forgot to use the constant here though, will update

I noticed that it is part of an interface further down the code so it's ok, didn't realise you had defined a new interface